src/components/Lists/index.jsx

```diff
@@ -7,6 +7,20 @@ import AllLists from './AllLists';
 export default function List() {
   const [newList, setNewList] = useState([]);
   const [allLists, setAllLists] = useState([]);
+  const [results, setResults] = useState([]);
+  const [idToReplace, setIdToReplace] = useState(null);
+
+  const replaceProduct = (newProduct) => {
```

Comment:
so whats happening to the id if we leave the editing mode? we should have an ability to set idToReplace back to null if we want to search a new item